app/api/events.py

```diff
@@ -1,90 +1,184 @@
-import datetime
-
+from datetime import datetime
+import os
+import shutil
 from fastapi import APIRouter, Response, Request, HTTPException, Depends
+from fastapi.datastructures import UploadFile
+from fastapi.param_functions import File
+from starlette.responses import FileResponse
+from uuid import uuid4, UUID
+
+from app.utils.validation import get_file_type, validate_image_file_type
 from ..db import get_database
 from bson.objectid import ObjectId
-from ..auth_helpers import authorize
-from ..models import Event, EventDB, AccessTokenPayload, Member
-from .util import crud_get_event_by_id
+from ..auth_helpers import authorize, role_required
+from ..models import Event, EventDB, AccessTokenPayload, EventInput, EventUpdate, Member
+from .utils import crud_get_event_by_id
 
 router = APIRouter()
 
 @router.post('/')
-def create_event(request: Request, newEvent: Event, token: AccessTokenPayload = Depends(authorize)):
+def create_event(request: Request, newEvent: EventInput, token: AccessTokenPayload = Depends(authorize)):
     db = get_database(request)
 
     # TODO better format handling and date date-time handling
     try :
-        event_date = datetime.datetime.strptime(str(newEvent.date), "%Y-%m-%d %H:%M:%S")
+        event_date = datetime.strptime(str(newEvent.date), "%Y-%m-%d %H:%M:%S")
     except ValueError:
         raise HTTPException(400, "Invalid date format")
 
-    if event_date < datetime.datetime.now():
+    if event_date < datetime.now():
         raise HTTPException(400, "Invalid date")
 
+    eid = uuid4().hex
     additionalFields = {
+        'eid' : eid,
         'participants' : [],
-        'posts' : []
+        'posts' : [],
     }
+    
     event = newEvent.dict()
     event.update(additionalFields)
 
     event = db.events.insert_one(event)
 
-    return {'_id' : str(event.inserted_id)}
+    return {'eid' : eid}
+
+@router.get('/')
+def get_all_event_ids(request: Request):
+    db = get_database(request) 
+
+    return [ str(event['eid']) for event in db.events.find() ]
+
+@router.get('/upcoming-events')
```

Comment:
Redundant use of `-events` in the name? since it's under the scope of events in the url with `/api/event`. What about `/api/event/upcoming`?

Reply:
agree :+1: